src/appleseed/foundation/math/sampling/qmcsamplingcontext.h

```diff
@@ -411,20 +379,46 @@ inline Vector<T, N> QMCSamplingContext<RNG>::next2(Tag<Vector<T, N>>)
             }
         }
     }
-    else if (m_mode == CMJMode)
+    else
     {
-        Vector<T, 2> sample_pos;
-        for (size_t i = 0; i < N; i += 2)
+        for (size_t i = 0; i < N; ++i)
+            v[i] = rand2<T>(m_rng);
+    }
+
+    ++m_instance;
+
+    return v;
+}
+
+template <typename RNG>
+template <typename T>
+inline std::vector<Vector<T, 2>> QMCSamplingContext<RNG>::next2(const size_t N)
+{
+	std::vector<Vector<T, 2>> v(N);
+
+    assert(m_sample_count == 0 || m_instance < m_sample_count);
```

Comment:
Why do you need m_sample_count?

Reply:
I did the same as in original next2 method

Reply:
It is nonsense here I think; in another next2 method actual number of calls is equal to sample count (you call it once per sample)